content/articles/working-with-styled-components-in-react/index.md

```diff
@@ -0,0 +1,504 @@
+
+```bash
+cd disney-landing-page
+
+```
+#### Installing styled-components
+Next, we install the `styled-components` and the `react-router-dom` libraries into our project through the commands below:
+
+```bash
+yarn add styled-components
+```
+
+```bash
+yarn add react-router-dom
+```
+#### Starting the development server
+To start the development server for the app, run either of the two commands on your terminal, depending on which package manager you are using. 
+
+```JavaScript
+yarn start
+```
+```JavaScript
+npm start
+```
+The expected result on your web browser is as shown:
+
+
+![Server](/engineering-education/working-with-styled-components/server start up.png)
+
+With everything set for our project, we can now open the project in a code editor and begin writing some code.
+
+Your expected app structure should look like this :
+
+![App structure](/engineering-education/working-with-styled-components/structure.png)
+
+
+#### Creating our Component 
+
+First, we need to create a folder to store our components. Within the *src* folder in your project structure, create a folder and name it *components*.
+
+![Components folder](/engineering-education/working-with-styled-components/component.png)
+
+In your newly created components folder, create two files and name one *Landing.js* and the other *Header.js*. Next, we will create our components and style them within these two files, as we will see shortly.
+
+Before we start working with the CSS-in-JS(styled-components), open the *Landing.js* file and the following code to create our first component: 
+
+We create the `Landing.js` component with the code below:
+
+```JavaScript
+import styled from "styled-components";// the styled components library we installed is imported here!
+// This component will render the landing page contents in container.
+const Landing = (props)=>{ // a functional component
+  return 
+     (<Container>
+         <Content>
+         <Content>
+     </Container>);
+}
+export default Landing;
+```
+
+To create our second component, open the `Header.js` file in the component folder. Add the following code to the file:
+
+```JavaScript
+
+import styled from "styled-components";
+
+// This component will render a Navbar before styling.
+const Header = (props)=>{
+    return (
+    <Nav>
+        Header
+    </Nav>
+    );
+}
+
+export default Header;
+```
+
+#### Routing a Component into the main App
+
+To get things started, let's open `App.js`  which is the base of our application. We replace all the content in it with the code below to create an App function that will load our styled-components and render them as the landing page.
+
+After building components/pages in your web application, you may need to expose and let your users navigate through them. To achieve this, you need a dedicated router. [React Router](https://www.freecodecamp.org/news/react-router-cheatsheet/) is a suite of navigational packages/components such as react-router-dom. 
```

Comment:
Explain react-router intuitively. You may direct the reader to an in-depth article on react-router. Navigational components such as react-router-dom does not convey information